src/include/distributed/worker_transaction.h

```diff
@@ -24,7 +24,8 @@
 typedef enum TargetWorkerSet
 {
 	NON_COORDINATOR_METADATA_NODES,
-	NON_COORDINATOR_NODES
+	NON_COORDINATOR_NODES,
+	ALL_DATA_NODES
```

Comment:
With this change, we're even increasing the complexity (#4016) by adding one more option :/ In fact, the name of the struct is `TargetWorkerSet` but  now we're adding the coordinator. In fact, `TargetWorkerSetNodeList ` seem to be designed/used only for worker nodes in the code

Is there another way to add the coordinator in a less controversial way? I mean, scoping the changes only to `DoRepartitionCleanup ()  / CreateTemporarySchemasForMergeTasks ()`?

In any case, we may consider renaming `ALL_DATA_NODES ` to `ALL_NODES ` as  we don't have a `data node ` concept in the code and lets not have it?

Reply:
It's probably a little nicer to pass function pointers instead of this enum, e.g. IsNodePrimaryWithShards

Reply:
>In fact, the name of the struct is TargetWorkerSet but now we're adding the coordinator.

I think this goes back to `What is a worker?` because assuming that we consider coordinator is also a worker when `shouldhaveshards=true` this enum will return it in `ALL_DATA_NODES` case. If we use `ALL_NODES` it seems a bit incorrect to me because I would expect `ALL_NODES` to return coordinator even if it is not added to the cluster as a worker.

So I guess maybe `ALL_SHARD_NODES`? This part also needs refactoring but would it be better to have that with the new structure which will be introduced for fixing #4016 (where we can also remove the worker concept)?